src/dirac_cwl_proto/job/__init__.py

```diff
@@ -67,39 +60,54 @@ def submit_job_client(
     - Start the jobs
     """
     # Validate the workflow
-    console.print(
-        "[blue]:information_source:[/blue] [bold]CLI:[/bold] Validating the job(s)..."
-    )
+    console.print("[blue]:information_source:[/blue] [bold]CLI:[/bold] Validating the job(s)...")
     try:
         task = load_document_by_uri(task_path)
     except ValidationException as ex:
-        console.print(
-            f"[red]:heavy_multiplication_x:[/red] [bold]CLI:[/bold] Failed to validate the task:\n{ex}"
-        )
+        console.print(f"[red]:heavy_multiplication_x:[/red] [bold]CLI:[/bold] Failed to validate the task:\n{ex}")
         return typer.Exit(code=1)
-    console.print(f"\t[green]:heavy_check_mark:[/green] Task {task_path}")
 
-    if metadata_path:
-        # Load the metadata (yaml fiel)
-        with open(metadata_path, "r") as file:
-            job_metadata = YAML(typ="safe").load(file)
-    else:
-        job_metadata = JobMetadataModel()
-    console.print("\t[green]:heavy_check_mark:[/green] Metadata")
+    console.print(f"\t[green]:heavy_check_mark:[/green] Task {task_path}")
 
+    job_metadata = JobMetadataModel()
     job_description = JobDescriptionModel(
         platform=platform,
         priority=priority,
         sites=sites,
     )
+    for hint in task.hints:
+        hint_class = hint["class"]
+        hint_stripped = {k: v for k, v in hint.items() if k != "class"}
+        if hint_class == "dirac:metadata":
+            console.print(f"Update metadata with:\n{hint_stripped}")
+            job_metadata = job_metadata.copy(update=hint_stripped)
+            continue
+        if hint_class == "dirac:description":
+            job_description = job_description.copy(update=hint_stripped)
```

Comment:
Correct me if I am wrong: 
- you kept the `job_metadata` and `job_description` input parameters of the CLI just to avoid breaking the current workflows and tests, right?
- but we actually don't need them anymore since this is now directly supported within CWL

Reply:
I believe we still need them, as the information in hints is our custom extension. CWL reference runner will just log a warning that it encountered unknown hints. An alternative would be to implement custom hints.